src/uct/ib/mlx5/rc/rc_mlx5_common.c

```diff
@@ -1088,9 +1078,15 @@ void uct_rc_mlx5_common_fill_dv_qp_attr(uct_rc_mlx5_iface_common_t *iface,
     }
 
 #ifdef HAVE_OOO_RECV_WRS
-    if (md->flags & UCT_IB_MLX5_MD_FLAG_DDP) {
+    if ((md->flags & UCT_IB_MLX5_MD_FLAG_DDP_RC) ||
+        (md->flags & UCT_IB_MLX5_MD_FLAG_DDP_DC)) {
```

Comment:
this might not work if we are in DC setup context, and only `(md->flags & UCT_IB_MLX5_MD_FLAG_DDP_RC)` is set. we will then set create_flags erroneously i think